### PR TITLE
Fix issue when compiling statics with multiple jobs

### LINKED ISSUE #38799: Theme compilation breaks when mysql host contains port information
### Preconditions and environment

- Magento 2.4.7
- PHP 8.3

### Steps to reproduce

1. Go to `env.php`
1. Set the port information to the host in the db connection. e.g. `db<'connection']['default']['host'] = 'localhost:3306'`
1. Execute  `./magento.sh setup:static-content:deploy`

### Expected result

The theme is deployed

### Actual result

Exception: 
`Port must be configured within host parameter (like localhost:3306 in /var/www/lehner_dev/src/www/vendor/magento/framework/DB/Adapter/Pdo/Mysql.php:464`

### Additional information

Possible solution:
``{{
diff --git a/DB/Adapter/Pdo/Mysql.php b/DB/Adapter/Pdo/Mysql.php
--- a/DB/Adapter/Pdo/Mysql.php	(revision 2dc0b58888392ed5304ac6ad02e36a70e6e490a2)
<ins></ins><ins> b/DB/Adapter/Pdo/Mysql.php	(date 1717577596939)
@@ -432,6 </ins>432,11 @@
             $this->parentConnections[] = $this->_connection;
             $this->_connection = null;
             $this->pid = getmypid();
<ins>            if (isset($this->*config['port']) && !str_contains($this->*config['host'], '/')) {
</ins>                if (!str*contains($this->*config['host'], ':'))
<ins>                    $this->*config['host'] .= ':' . $this->*config['port'];
</ins>                unset($this->_config['port']);
+            }
         }
     }
}}``

### Release note

*No response*

### Triage and priority

- [ ] Severity: **S0** *- Affects critical data or functionality and leaves users without workaround.*
- [ ] Severity: **S1** *- Affects critical data or functionality and forces users to employ a workaround.*
- [ ] Severity: **S2** *- Affects non-critical data or functionality and forces users to employ a workaround.*
- [ ] Severity: **S3** *- Affects non-critical data or functionality and does not force users to employ a workaround.*
- [ > Severity: **S4** *- Affects aesthetics, professional look and feel, "quality" or "usability".*

### PR DESCRIPTION
**Pull request by mrodespin**
Patch to fix open Magento issue https://github.com/magento/magento2/issues/38799
- Fixes https://github.com/magento/magento2/issues/38799

### CODE REVIEW

**Comment by jellesiderius** [comment]
Can confirm that this fixes my issues.

**Comment by monteshot** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]

File: lib/internal/Magento/Framework/DB/Adapter/Pdo/Mysql.php [hunk 1]
```diff
@@ -432,6 +432,12 @@ private function avoidReusingParentProcessConnection()
             $this->parentConnections[] = $this->_connection;
             $this->_connection = null;
             $this->pid = getmypid();
+
+            // Reset config host to avoid issue with multiple connections
+            if (!empty($this->_config['port'])) {
```
Comment (engcom-Hotel) [inline lib/internal/Magento/Framework/DB/Adapter/Pdo/Mysql.php]: Can we add a check here if the port is not already set with localhost then only perform this action:
```suggestion
            if (!empty($this->_config['port']) && strpos($this->_config['host'], ':') === false) {
```

**Review by engcom-Hotel — CHANGES_REQUESTED** [review]
Hello @mrodespin,

Thanks for the collaboration!

Please refer to my review comment below.

Thanks

**Review by ihor-sviziev — APPROVED** [review]

**Comment by engcom-Bravo** [comment]
@magento run all tests

**Review by engcom-Hotel — APPROVED** [review]

**Comment by engcom-Bravo** [comment]
Hi @mrodespin,

Thanks for the collaboration & contribution!

### :heavy_check_mark: QA Passed


**Preconditions:**

- Install fresh Magento **`2.4-develop`**

**Steps to reproduce**


- Go to env.php
- Set the port information to the host in the db connection. e.g. db['connection']['default']['host'] = 'localhost:3306'
- Execute ./magento.sh setup:static-content:deploy


**Before:** :heavy_multiplication_x: 


Screenshot 2024-08-22 at 12 23 47



**After:** :heavy_check_mark: 


Screenshot 2024-08-22 at 13 32 33



Builds are failed. Hence, moving this PR to **Extended Testing**.

Thanks

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by engcom-Charlie** [comment]
The Functional B2B and EE failures in recent 2 builds are known issues and not related to the PR changes hence moving this PR to Merge In Progress.

**Functional B2B**
**Run 1:**  
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/38842/1848ea78770547fa01e0c248208b0494/Functional/allure-report-b2b/index.html#categories/e89a2ef77b192fa659586970627a6e4a/3fee4ef775faff28/

image


**Run 2:** 
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/38842/fb6fb189ab0db3ae415a94d68f159b02/Functional/allure-report-b2b/index.html#categories/e89a2ef77b192fa659586970627a6e4a/f6d6cdc83faabb49/

image


**Functional EE**
**Run 1:** 
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/38842/193e47cecbabff724bf8865e273822b9/Functional/allure-report-ee/index.html#categories

image

**Run 2:**
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/38842/0923be2ff0cf4098bebd1a906fdf7cf7/Functional/allure-report-ee/index.html#categories/5c3e33efaf655a884aa27c9147cb6ca1/17474efe84ca217c/

image


**Known issues :**
EE: StorefrontCreateOrderAllQuantityGroupedProductOptionDefaultStockTest: [#ACQE-6331](https://jira.corp.adobe.com/browse/ACQE-6331)
B2B: AdminCreateDraftQuoteFromQuoteGridPageTest: [#ACQE-6309](https://jira.corp.adobe.com/browse/ACQE-6309)

**Pull request merged**